File: LanguageConvertor/Tests/Output/Java/FromCSharp.java
package Output.Java;

import Output.Java.Example.*;

public class FromCSharp extends Base implements IInterface
{
    public int number = 0;
    public float weight;
    private String stringPropertyBackingField;
    private int intPropertyBackingField;
    
    public FromCSharp()
    {
        StringProperty = "NULL";
        IntProperty = 0;
    }
    
    public FromCSharp(String str, int integer)
    {
        StringProperty = str;
        IntProperty = integer;
    }
    
    public void method()
    {
    }
    
    public void func3(Object obj)
    {
    }
    
    public String getStringProperty()
    {
        return stringPropertyBackingField;
    }
    
    public int getIntProperty()
    {
        return intPropertyBackingField;
    }
    
    private static int add(int[] numbers, int count)
    {
        return 0;
    }
    
    void explode(boolean sure)
    {
    }
    
    @Override
    protected void func1(int obj)
    {
    }
    
    @Override
    protected void func2(String obj)
    {
    }
    
    protected void setStringProperty(String value)
    {
        stringPropertyBackingField = value;
    }
    
}

File: LanguageConvertor/Tests/Output/Python/FromCSharp.py


from Output import *

@dataclass
class FromCSharp(Base, IInterface):
    number = 0
    weight: float
    stringPropertyBackingField: str
    intPropertyBackingField: int
    
    def fromCSharp(self) -> None:
        StringProperty = "NULL";
        IntProperty = 0;
    
    def fromCSharp(self, str: str, integer: int) -> None:
        StringProperty = str;
        IntProperty = integer;
    
    def method(self) -> None:
        pass
    
    def func3(self, obj: object) -> None:
        pass
    
    def getStringProperty(self) -> str:
        return stringPropertyBackingField
    
    def getIntProperty(self) -> int:
        return intPropertyBackingField
    
    def add(self, numbers: list[int], count: int) -> int:
        return 0;
    
    def explode(self, sure: bool) -> None:
        pass
    
    def func1(self, obj: int) -> None:
        pass
    
    def func2(self, obj: str) -> None:
        pass
    
    def setStringProperty(self, value: str) -> None:
        self.stringPropertyBackingField = value
    

